### DISCUSSION TITLE
Disable / Skip LCD Displays (i.E. Raidmode, Link)

### DISCUSSION THREAD

**Discussion by FleX246** [1000001101000/Python_buffalo_libmicon#7] [General]
Is there a possibility when I activate the mode that the displays are switched automatically (or switch via Button) the displays "raidmode", "diskcap" and "linkspeed" are skipped ?

I would like to switch between "buffer0" (the user-defined display), "hostname & ip" and "date & time" either automatically or via a button (I'm not sure about that yet).

thanks!
FleX246

**Comment by 1000001101000** [comment]
Good to hear from you!

Some of the device specific example scripts run through how to do what you are describing thought the part about which screens are actually active in the cycle aren't very clear. 

Likewise, the documentation of the protocol didn't call out how that work either. I've added some notes to the protocol documentation to make it a little more clear:
https://buffalonas.miraheze.org/wiki/MiconV2_Interface

I believe this script does what you're after (at least on my TS3400D), it's excerpts from ts3000_examples.py with the display bitmask tweaked. 

I confirmed both cycling via display button and the 3sec auto mode work (uncomment the desired line). 

```
#!/usr/bin/python3

import libmicon
import socket
import time

test = libmicon.micon_api("/dev/ttyS1")

##update the date for the lcd display
test.set_lcd_date()

##set the ip address for the lcd display
s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
s.connect(('10.255.255.255', 1))
host_ip = s.getsockname()[0]
s.close()
host_name = socket.gethostname()
test.set_lcd_buffer_short(libmicon.lcd_set_ipaddress,host_ip)
test.set_lcd_buffer_short(libmicon.lcd_set_hostname,host_name)

##set custom lcd message
test.set_lcd_buffer(libmicon.lcd_set_buffer0,"Terastation ARM","   Debian")

##enable just the messages that we've configured so far.
test.send_write_cmd(1,libmicon.lcd_set_dispitem,0x34)
test.send_write_cmd(1,libmicon.lcd_set_dispitem_ex,0x00)

##configure to cycle through displays via the display button
test.send_write_cmd(0,libmicon.lcd_changemode_button)

##configure to cycle through displays every ~3 seconds
#test.send_write_cmd(0,libmicon.lcd_changemode_auto)
```

**Comment by FleX246** [comment]
I thank you very much for the very quick answer!

But I obviously destroyed my Buffalo NAS (TS-QVHL) permanently.

I caused a short circuit on the power supply and now the power supply and the mainboard are defective.
Both show no more reactions.

A used power supply unit can be found on the Internet, but I can't find a motherboard suitable for this NAS.

Fortunately, the hard drives are still OK :-D

I don't know yet whether the new NAS will be a Buffalo again.
If it becomes one, I'll definitely have another look here.

Anyway, thanks again for the quick response and the work you are doing here for Buffalo NAS owners.

**Comment by 1000001101000** [comment]
How did that happen?

Maybe if you're lucky it's just the PSU fuse or something fixable/replaceable. I have some pictures on the wiki if that helps:
https://buffalonas.miraheze.org/wiki/Terastation_TS-QVHL

I also have a spare TS-QVHL if you live on the same continent and want to work out a parts swap or something.

**Comment by FleX246** [comment]
I did not connect the PE / protective conductor (in German it calls "Schutzleiter") correctly and this then touched the primary power supply side.
This led to the fact that some components on the power supply (at least 1 thermistor, 1 resistor (SMD), 1 diode (SMD) and 1 capacitor (SMD) burned out.
I tried to start the mainboard with a "normal PC power supply" but it is also without function.

I saw at least one IC on the mainboard that was also damaged.

Regarding your TS-QVHL.  Are the mainboard and power supply still working?  I come from Germany...

**Comment by 1000001101000** [comment]
Sounds like you've already looked into it beyond what I would know to do. Sorry to hear about that. 

I believe mine is fully functional. Unfortunately I'm in the USA.

**Comment by FleX246** [comment]
Would you want to send it to Germany?
What price would you want for the NAS and what would the shipping cost?
